api/src/pcapi/scripts/clean_offer_titles_with_eans/main.py

```diff
@@ -0,0 +1,242 @@
+            )
+        )
+
+    return rows
+
+
+def run() -> None:
+    while True:
+        rows = get_offers_with_ean_inside_title()
+        if not rows:
+            break
+
+        parse_offers(rows)
+
+
+def parse_offers(rows: Collection[OfferEanQueryRow]) -> None:
+    for chunk in get_chunks(rows, chunk_size=2_000):
+
+        unknown_offer_rows = set()
+        gcu_incompatible_offer_rows = set()
+        legit_offer_rows = set()
+
+        for offer_row in chunk:
+            if not offer_row.exists:
+                unknown_offer_rows.add(offer_row)
+            elif offer_row.gcu_compatibility != GcuCompatibilityType.COMPATIBLE.value:
+                gcu_incompatible_offer_rows.add(offer_row)
+            else:
+                legit_offer_rows.add(offer_row)
+
+        update_unknown_offers(unknown_offer_rows)
+        update_gcu_incompatible_offers(gcu_incompatible_offer_rows)
+        update_legit_offers(legit_offer_rows)
+
+
+def retry_and_log(func: Callable) -> Callable:
+    @atomic()
+    def retry_one_chunk_at_a_time(offer_rows: Collection[OfferEanQueryRow]) -> None:
+        chunk_size = len(offer_rows) // 5
+        chunk_size = max(chunk_size, 1)
+
+        for chunk in get_chunks(offer_rows, chunk_size=chunk_size):
+            try:
+                with atomic():
+                    func(chunk)
+            except Exception as exc:  # pylint: disable=broad-exception-caught
+                if chunk_size == 1:
+                    row = chunk[0]
+                    msg = "[%s][%s] could not handle offer #%s with product #%s (ean: %s)"
+                    logger.info(msg, str(exc), func.__name__, row.id, row.product_id, row.ean)
+                else:
+                    retry_one_chunk_at_a_time(chunk)
+                continue
+
+    def inner(offer_rows: Collection[OfferEanQueryRow]) -> bool:
+        try:
+            func(offer_rows)
+        except Exception:  # pylint: disable=broad-exception-caught
+            retry_one_chunk_at_a_time(offer_rows)
+            return False
+        return True
+
+    return inner
+
+
+@atomic()
+@retry_and_log
+def update_unknown_offers(offer_rows: Collection[OfferEanQueryRow]) -> None:
+    reject_offers(offer_rows)
+
+
+@atomic()
+@retry_and_log
+def update_gcu_incompatible_offers(offer_rows: Collection[OfferEanQueryRow]) -> None:
+    reject_offers(offer_rows)
+
+
+@atomic()
+@retry_and_log
+def update_legit_offers(offer_rows: Collection[OfferEanQueryRow]) -> None:
+    ids = {row.id for row in offer_rows}
+    legit_offers = Offer.query.filter(Offer.id.in_(ids))
+
+    offer_to_product = {row.id: row for row in offer_rows}
+
+    with atomic():
+        for offer in legit_offers:
+            offer.name = offer_to_product[offer.id].product_name
+
+            if offer_to_product[offer.id].product_json_data:
+                extra_data = cast(str, offer_to_product[offer.id].product_json_data)
+                offer.extraData = json.loads(extra_data)
```

Comment:
On pourrait simplifier cette partie là, si `product_json_data` était directement un `dict` et non une `str`.
```suggestion
            if offer_to_product[offer.id].product_json_data:
                offer.extraData = offer_to_product[offer.id].product_json_data
```